src/Drivers/Gd/Encoders/GifEncoder.php

```diff
@@ -43,7 +43,11 @@ protected function encodeAnimated(ImageInterface $image): EncodedImage
             );
         }
 
-        $builder->setLoops($image->loops());
+        try {
+            $builder->setLoops($image->loops());
```

Comment:
setLoops may throw an exception if the loop is empty.

I try catch to transform this to a custom exception (and respect the `@throws` phpdoc)